packages/core/src/modules/oob/__tests__/OutOfBandMessage.test.ts

```diff
@@ -0,0 +1,87 @@
+import type { ValidationError } from 'class-validator'
+
+import { JsonEncoder } from '../../../utils/JsonEncoder'
+import { JsonTransformer } from '../../../utils/JsonTransformer'
+import { OutOfBandMessage } from '../messages/OutOfBandMessage'
+
+describe('OutOfBandMessage', () => {
+  describe('toUrl', () => {
+    test('encode the message into the URL containg the base64 encoded invitation as the oob query parameter', async () => {
+      const domain = 'https://example.com/ssi'
+      const json = {
+        '@type': 'https://didcomm.org/out-of-band/1.1/invitation',
+        services: ['did:sov:LjgpST2rjsoxYegQDRm7EL'],
+        '@id': '69212a3a-d068-4f9d-a2dd-4741bca89af3',
+        label: 'Faber College',
+        goal_code: 'issue-vc',
+        goal: 'To issue a Faber College Graduate credential',
+        handshake_protocols: ['https://didcomm.org/didexchange/1.0', 'https://didcomm.org/connections/1.0'],
+      }
+      const invitation = JsonTransformer.fromJSON(json, OutOfBandMessage)
+      const invitationUrl = invitation.toUrl({
+        domain,
+      })
+
+      expect(invitationUrl).toBe(`${domain}?oob=${JsonEncoder.toBase64URL(json)}`)
+    })
+  })
+
+  describe('fromUrl', () => {
+    test('decode the URL containing the base64 encoded invitation as the oob parameter into an `OutOfBandMessage`', async () => {
+      const invitationUrl =
+        'http://example.com/ssi?oob=eyJAdHlwZSI6Imh0dHBzOi8vZGlkY29tbS5vcmcvb3V0LW9mLWJhbmQvMS4xL2ludml0YXRpb24iLCJAaWQiOiI2OTIxMmEzYS1kMDY4LTRmOWQtYTJkZC00NzQxYmNhODlhZjMiLCJsYWJlbCI6IkZhYmVyIENvbGxlZ2UiLCJnb2FsX2NvZGUiOiJpc3N1ZS12YyIsImdvYWwiOiJUbyBpc3N1ZSBhIEZhYmVyIENvbGxlZ2UgR3JhZHVhdGUgY3JlZGVudGlhbCIsImhhbmRzaGFrZV9wcm90b2NvbHMiOlsiaHR0cHM6Ly9kaWRjb21tLm9yZy9kaWRleGNoYW5nZS8xLjAiLCJodHRwczovL2RpZGNvbW0ub3JnL2Nvbm5lY3Rpb25zLzEuMCJdLCJzZXJ2aWNlcyI6WyJkaWQ6c292OkxqZ3BTVDJyanNveFllZ1FEUm03RUwiXX0K'
+
+      const invitation = await OutOfBandMessage.fromUrl(invitationUrl)
+      const json = JsonTransformer.toJSON(invitation)
+
+      expect(json).toEqual({
+        '@type': 'https://didcomm.org/out-of-band/1.1/invitation',
```

Comment:
```suggestion
        '@type': 'https://didcomm.org/out-of-band/1.0/invitation',
```

Reply:
Hmm, I'm confused why the header in Aries RFC says version 1.1? 🤔 

Aries RFC 0434: Out-of-Band Protocol 1.1

Reply:
Ah my bad. You're right it should be 1.1. But then we should use 1.1 everywhere. Other messages are using 1.0, also the handshake_protocols array

Reply:
You're right that `HandshakeReuseMessage` is using `1.0` and should use `1.1`, I'll fix that. I'm not sure what you mean by version in other messages and `handshake_protocols` array. The `handshake_protocols` array contains just what's supported by the framework and `did-exchange` and `connections` RFCs define version `1.0`.